### PR TITLE
Rotation Supported

### PR DESCRIPTION
**Pull request by tuanna222**
- Adding SATPhysics Component. Allow switching bettween SATPhysics and EntityPhysics. Default: EntityPhysics
- Adding rotate and setAnchor (pivot point) function on Entity
- Supporting detect collision with rotation on any pivot
- Changing response object on EntityPool

### CODE REVIEW

**Comment by rogueSkib** [comment]
This is great! Thanks @tuanna222 ! I'm going to make a few edits / updates and push to this branch. After that, if they work for you, I'll merge and tag :+1:

File: src/Entity.js [hunk 1]
```diff
@@ -214,22 +259,9 @@ exports = Class(function() {
 	 *         ~ returns the total distance moved by the two entities
 	 */
 
-	this.collidesWith = function(entity) {
-		return this.physics.collide(this, entity);
-	};
-
-	this.resolveCollidingStateWith = function(entity) {
```
Comment (sjrct) [inline src/Entity.js:221]: Why remove the `resolveCollidingStateWith` function?

Comment (tuanna222) [inline src/Entity.js:221]: Thank @sjrct , this function was restored to support EntityPhysics.
If you use SATPhysics, you should use `response.overlap`.

Comment (rogueSkib) [inline src/Entity.js:221]: @tuanna222 SATPhysics should also support this API by using `response.overlap` as you mentioned. Can you change the call to `resolveCollidingStateWith` to accept an optional `response` object, which gets passed into physics? You'll need to add the missing function `resolveCollidingState` to SATPhysics. The other missing functions `resolveCollidingCircles`, `resolveCollidingCircleRect`, and `resolveCollidingRects` are specific to EntityPhysics; I imagine the SATPhysics counterparts might be similar, except replace Rect with Polygon?

In general, I marked the physics functions that are required by the Entity API with the tag `* ~ REQUIRED for Entity`.

I started a merger yesterday but have a lot of other stuff I'm working on. So far, I've repaired the README file and I'm working on removing most physics specific logic you added to Entity.js into SATPhysics and EntityPhysics. I'll push my code to the `merge-sat` branch, but it's still a work-in-progress, and not ready to merge. At least you can see the direction I was heading.

You will also want to match our style when you edit code in one of the original files. Adding external files in their own style is fine though, like SAT.js.

Comment (tuanna222) [inline src/Entity.js:221]: I did some fixes on `merge-sat` branch and also formatted the code style. Thank @rogueSkib !

**Comment by lunarraid** [comment]
Closing in favor of https://github.com/gameclosure/devkit-entities/pull/3

**Pull request closed without merging**